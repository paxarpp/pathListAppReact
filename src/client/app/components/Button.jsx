import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ handler, styleButton, children }) => {
  return (
    <a onClick={handler} className={styleButton}>
      {children}
    </a>
  );
};
Button.propTypes = {
  handler: PropTypes.func,
  styleButton: PropTypes.string,
  children: PropTypes.string
};
export default Button;
