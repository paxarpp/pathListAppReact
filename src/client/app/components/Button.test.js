import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('Button has the class', () => {
  const component = renderer.create(
    <Button handler={() => {}} styleButton="addCar">
      создать
    </Button>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button has the class , no children', () => {
  const component = renderer.create(<Button handler={() => {}} styleButton="addCar" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
