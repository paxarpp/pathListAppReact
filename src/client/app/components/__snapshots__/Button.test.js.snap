// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has the class , no children 1`] = `
<a
  className="addCar"
  onClick={[Function]}
/>
`;

exports[`Button has the class 1`] = `
<a
  className="addCar"
  onClick={[Function]}
>
  создать
</a>
`;
